### ISSUE TITLE
How Can I Disable Foreign Key?

### ISSUE THREAD

**Issue by linxixuan** [prisma/prisma#8108]
## Problem

My DB can't use foreign key，If I specify the relation, Prisma will create foreign constraint  automatically.

## Suggested solution

I wonder how can I disable the foreign key constraint? All I need is the relation between two tables 

So, is it possible 

## Some Info

here is the schema and sql created by prisma

```
model Post {
  id          Int           @id @default(autoincrement())
  updateAt    DateTime      @default(now())
  title       String
  type        String
  url         String        @unique
  tags        String
  authorId    Int?
  likes       Int           @default(0)
  isDelete    Boolean       @default(false)
  user        User?         @relation(fields: [authorId], references: [id])
  weeklyPosts WeeklyPosts[]

  @@index([authorId], name: "authorId")
}
```

```sql
CREATE TABLE `Post` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `updateAt` datetime(3) NOT NULL DEFAULT CURRENT_TIMESTAMP(3),
  `title` varchar(191) COLLATE utf8mb4_unicode_ci NOT NULL,
  `type` varchar(191) COLLATE utf8mb4_unicode_ci NOT NULL,
  `url` varchar(191) COLLATE utf8mb4_unicode_ci NOT NULL,
  `tags` varchar(191) COLLATE utf8mb4_unicode_ci NOT NULL,
  `authorId` int(11) DEFAULT NULL,
  `likes` int(11) NOT NULL DEFAULT '0',
  `isDelete` tinyint(1) NOT NULL DEFAULT '0',
  PRIMARY KEY (`id`),
  UNIQUE KEY `Post.url_unique` (`url`),
  KEY `authorId` (`authorId`),
  CONSTRAINT `post_ibfk_1` FOREIGN KEY (`authorId`) REFERENCES `user` (`id`) ON DELETE SET NULL ON UPDATE CASCADE
) ENGINE=InnoDB AUTO_INCREMENT=2 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
```

**Comment by tomhoule** [comment]
Hi @linxixuan — what database are you using? We are discussing how we tackle this internally and want to work on an option to make migrate ignore relations (-> create no foreign keys) soon. In the meantime, you can add `previewFeatures = ["planetScaleMode]` to your generator and `planetScaleMode = true` in your datasource and it should do what you want. But these options are for sure going to change soon.

**Comment by linxixuan** [comment]
DB is mysql, our DBA suggest us do not use foreign key for some performance reason ;)

**Comment by linxixuan** [comment]
@tomhoule hello, your solution is not work. Here is my env

## bug
![image](https://user-images.githubusercontent.com/3753428/124593461-425dcd80-de91-11eb-8e84-9e05f958bbb2.png)


## my code

```
generator client {
  provider = "prisma-client-js"
  previewFeatures = ["planetScaleMode"]
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
  planetScaleMode = true
}
```


## env
```bash
$ npx prisma -v
prisma               : 2.23.0
@prisma/client       : 2.23.0
Current platform     : darwin
Query Engine         : query-engine adf5e8cba3daf12d456d911d72b6e9418681b28b (at ../../node_modules/@prisma/engines/query-engine-darwin)
Migration Engine     : migration-engine-cli adf5e8cba3daf12d456d911d72b6e9418681b28b (at ../../node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine : introspection-core adf5e8cba3daf12d456d911d72b6e9418681b28b (at ../../node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary        : prisma-fmt adf5e8cba3daf12d456d911d72b6e9418681b28b (at ../../node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash : adf5e8cba3daf12d456d911d72b6e9418681b28b
Studio               : 0.393.0
```

**Comment by tomhoule** [comment]
Ah, I think your prisma version is a bit too old, it probably was released around 2.24

**Comment by thebiglabasky** [comment]
@linxixuan Did you try with the latest version to confirm we can close this issue?
Thank you

**Comment by janpio** [comment]
Can you please open a new issue @sam-bunger? I think your problem is different. Thanks.

**Comment by sam-bunger** [comment]
Hey @janpio, the problems are different, but the ability to disable foreign key constraints would solve both of them.

**Comment by janpio** [comment]
The current solution is described above: https://github.com/prisma/prisma/issues/8108#issuecomment-874575155

**Comment by bryanc208** [comment]
Hi @sam-bunger – did you manage to figure out how to solve your use-case? I'm hoping to achieve the same where I have a Postgres DB and need to set up a relation that does not have a foreign key constraint.

**Comment by sam-bunger** [comment]
Hey @bryanc208, I couldn't figure it out. We ended up deciding to stop using prisma for our project, so I haven't had the chance to look around and test if this solution really works, but from this looks of it this PR may have fixed the issue: prisma/prisma-engines#2221

**Comment by Arthur-CWW** [comment]
for anyone searching for this and using planetscale in the future [foreign key mode](https://www.prisma.io/docs/guides/database/using-prisma-with-planetscale#how-to-enable-emulation-of-referential-integrity) .
```
datasource db {
  provider     = "mysql"
  url          = env("DATABASE_URL")
  relationMode = "prisma"
}
```
